qiskit/opflow/converters/pauli_basis_change.py

```diff
@@ -346,7 +346,9 @@ def get_diagonalizing_clifford(self, pauli: Union[Pauli, PauliOp]) -> OperatorBa
         y_to_x_origin = tensorall(
             [S if has_y else I for has_y in reversed(np.logical_and(pauli.x, pauli.z))]
         ).adjoint()
-        x_to_z_origin = tensorall([H if has_x else I for has_x in reversed(pauli.x)])
+        x_to_z_origin = tensorall(  # pylint: disable=assignment-from-no-return
+            [H if has_x else I for has_x in reversed(pauli.x)]
+        )
```

Comment:
Huh, this smells like a Pylint bug, since `tensorall` is super explicitly typed to be callable right above this diff.

Reply:
I also think it's strange. What's really fishy is it only happened in CI, and not locally. The only idea I had for that is my local is using Python 3.11.

Reply:
This isn't the first time I've heard of pylint behaving differently in different versions of Python, though it's pretty weird because `astroid` and `pylint` are both pure-Python packages.